cmake/compilers/Clang.cmake

```diff
@@ -24,11 +30,17 @@ if (NOT TBB_STRICT AND COMMAND tbb_remove_compile_flag)
     tbb_remove_compile_flag(-Werror)
 endif()
 
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "(amd64)|(AMD64)|(x86_64)")
```

Comment:
Seems the expression can be a bit simplified:
`"(amd64|AMD64|x86_64)"`

Reply:
Clang+LLD on Windows defaults to outputting big object files by default, so we don't need the flag.
Speaking of object files, currently gcc fails to compile `test_eh_flow_graph` on Windows, due to the linker complaining about string table overflow. We might have to turn on optimizations for the test, to reduce the symbol length.